api/product.rb

```diff
@@ -91,6 +91,24 @@ def set_properties_based_on_version(version)
     end
     # rubocop:enable Naming/AccessorMethodName
 
+    def exists_at_version_or_lower(name)
+      return true if @versions.nil?
+      name ||= 'ga'
+      return false unless Version::ORDER.include?(name)
+      (0..Version::ORDER.index(name)).each do |i|
+        return true if exists_at_version(Version::ORDER[i])
+      end
+      false
+    end
+
+    def exists_at_version(name)
+      return true if @versions.nil?
```

Comment:
Does this mean if a product doesn't define any versions it exists in all of them? Should it be false?

Reply:
This is to get around the case where _bundle (which is just a chef/puppet thing) wasn't able to get generated because it has no versions defined. It was either this or special-case bundle completely. Any preference?

Reply:
It looks like you can't remove the versions block from a 'normal' product anyway and I'm not totally sure why bundle gets to be versionless. But I don't really understand the bundle products to begin with.

I think that resources won't work without a version block defining the base url for a particular version so I think it's fine as is.

Reply:
Thanks! I added a comment to explain.